src/gui/shell.cpp

```diff
@@ -292,6 +298,12 @@ void Shell::handleResize(uint64_t cols, uint64_t rows)
 		}
 		m_image.swap(new_image);
 		updateGeometry();
+		// Layout recalculation is done in next event cycle, but we start
+		// rendering right after emitting the neovimResized() signal.
+		if (parentWidget() && parentWidget()->layout()) {
+			parentWidget()->layout()->invalidate();
+			parentWidget()->layout()->activate();
+		}
```

Comment:
Yes, in next event cycle, so `sizeHint()` is wrong in slot connected to `neovimResize()` that is emitted on the next line.